aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/UserPoolConfiguration.kt

```diff
@@ -62,24 +64,48 @@ data class UserPoolConfiguration internal constructor(val builder: Builder) {
         configJson: JSONObject? = null
     ) {
         var region: String? = DEFAULT_REGION
+        var endpoint: String? = null
         var poolId: String? = null
         var appClientId: String? = null
         var appClientSecret: String? = null
 
         init {
             configJson?.run {
                 region = optString(Config.REGION.key).takeUnless { it.isNullOrEmpty() }
+                endpoint = validateEndpoint(optString(Config.ENDPOINT.key).takeUnless { it.isNullOrEmpty() })
                 poolId = optString(Config.POOL_ID.key).takeUnless { it.isNullOrEmpty() }
                 appClientId = optString(Config.APP_CLIENT_ID.key).takeUnless { it.isNullOrEmpty() }
                 appClientSecret = optString(Config.APP_CLIENT_SECRET.key).takeUnless { it.isNullOrEmpty() }
             }
         }
 
         fun region(region: String) = apply { this.region = region }
+        fun endpoint(endpoint: String) = apply { this.endpoint = validateEndpoint(endpoint) }
         fun poolId(poolId: String) = apply { this.poolId = poolId }
         fun appClientId(appClientId: String) = apply { this.appClientId = appClientId }
         fun appClientSecret(appClientSecret: String) = apply { this.appClientSecret = appClientSecret }
         fun build() = UserPoolConfiguration(this)
+
+        @Throws(AuthException::class)
+        private fun validateEndpoint(endpoint: String?): String? {
```

Comment:
Please write a test for this with known endpoints that should work and those that should not.

Reply:
Yes just added them